### PR TITLE
fix: invalid prices on grouped and configurable product

### PR DESCRIPTION
**Pull request by bartoszherba**
## Description
 fix: invalid prices on grouped and configurable product
    - fix issue with initial grouped product price resolving
    - add configurable product price calculation

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by sethidden — APPROVED** [review]

**Review by sethidden — CHANGES_REQUESTED** [review]
temporary block - reviewing

**Review by sethidden — DISMISSED** [review]
Left some optional comments

File: packages/theme/modules/catalog/pricing/__tests__/getConfigurableProductPriceCommand.spec.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import ConfigurableProductWithSelectionMock from '~/modules/catalog/pricing/__tests__/mock/ConfigurableProductWithSelection.mock';
+
+describe('getConfigurableProductPriceCommand', () => {
+  it('Resolve price from the parent if no variant is selected', () => {
+    // @ts-ignore for the simplified mock structure
+    expect(getConfigurableProductPriceCommand(ConfigurableProductWithoutSelectionMock)).toEqual(20);
+  });
+
+  it('Resolve price from the variant if variant is selected', () => {
+    // @ts-ignore for the simplified mock structure
```
[the remark this answers is not in the document]
Comment (bartoszherba) [reply packages/theme/modules/catalog/pricing/__tests__/getConfigurableProductPriceCommand.spec.ts:12]: Ok, fixed.

File: packages/theme/modules/catalog/pricing/getGroupedProductPriceCommand.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import type { GroupedProduct, ProductInterface } from '~/modules/GraphQL/types';
+import { getPrice as getProductPrice } from '~/modules/catalog/product/getters/productGetters';
+
+export function getGroupedProductPriceCommand(product: GroupedProduct): number {
+  const evalProductPrice = (p: ProductInterface) => {
+    const { regular, special } = getProductPrice(p);
+
+    return regular > special ? regular : special;
+  };
```
[the remark this answers is not in the document]
Comment (sethidden) [reply packages/theme/modules/catalog/pricing/getGroupedProductPriceCommand.ts:9]: suits me ¯\_(ツ)_/¯

File: packages/theme/modules/catalog/product/enums/ProductTypeEnum.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export type ProductTypeEnum = 'SimpleProduct' | 'ConfigurableProduct' | 'BundleProduct' | 'GroupedProduct';
```
[the remark this answers is not in the document]
Comment (sethidden) [reply packages/theme/modules/catalog/product/enums/ProductTypeEnum.ts:1]: this is just about the name, using unions or enums is fine

**Pull request merged**